room/runtime/src/main/java/androidx/room/SQLiteCopyOpenHelper.java

```diff
@@ -177,7 +192,12 @@ private void copyDatabaseFile(File destinationFile) throws IOException {
         if (mCopyFromAssetPath != null) {
             input = Channels.newChannel(mContext.getAssets().open(mCopyFromAssetPath));
         } else if (mCopyFromFile != null) {
-            input = new FileInputStream(mCopyFromFile).getChannel();
+            final FileInputStream fileInputStream = new FileInputStream(mCopyFromFile);
+            if (mIsCopyFromFileGzipped) {
+                input = Channels.newChannel(new GZIPInputStream(fileInputStream));
```

Comment:
only opted for gzip because it's simple - zip itself will be a lot of code to loop through, get and extract zip entries, and so on which is probably not necessary?